### ISSUE TITLE
Error when configuring filters

### ISSUE THREAD

**Issue by anjackson** [aadel/sqlalchemy-solr#2]
Thanks for this module! I've been trying it out with Apache Superset 0.37.2 and this error cropped up when trying to set up a filter:

```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 747, in __getattr__
    return getattr(self.comparator, key)
AttributeError: 'Comparator' object has no attribute 'text'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/superset/viz.py", line 503, in get_df_payload
    df = self.get_df(query_obj)
  File "/usr/local/lib/python3.7/site-packages/superset/viz.py", line 258, in get_df
    self.results = self.datasource.query(query_obj)
  File "/usr/local/lib/python3.7/site-packages/superset/connectors/sqla/models.py", line 1193, in query
    query_str_ext = self.get_query_str_extended(query_obj)
  File "/usr/local/lib/python3.7/site-packages/superset/connectors/sqla/models.py", line 679, in get_query_str_extended
    sql = self.database.compile_sqla_query(sqlaq.sqla_query)
  File "/usr/local/lib/python3.7/site-packages/superset/models/core.py", line 413, in compile_sqla_query
    sql = str(qry.compile(engine, compile_kwargs={"literal_binds": True}))
  File "<string>", line 1, in <lambda>
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 481, in compile
    return self._compiler(dialect, bind=bind, **kw)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 487, in _compiler
    return dialect.statement_compiler(dialect, self, **kw)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/compiler.py", line 592, in __init__
    Compiled.__init__(self, dialect, statement, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/compiler.py", line 322, in __init__
    self.string = self.process(self.statement, **compile_kwargs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/compiler.py", line 352, in process
    return obj._compiler_dispatch(self, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 96, in _compiler_dispatch
    return meth(self, **kw)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/compiler.py", line 2164, in visit_select
    text, select, inner_columns, froms, byfrom, kwargs
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/compiler.py", line 2263, in _compose_select_body
    t = select._whereclause._compiler_dispatch(self, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/visitors.py", line 96, in _compiler_dispatch
    return meth(self, **kw)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy_solr-0.1.dev0-py3.7.egg/sqlalchemy_solr/base.py", line 111, in visit_clauselist
    parser.parse(c.right.text)
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/sql/elements.py", line 758, in __getattr__
    replace_context=err,
  File "/usr/local/lib/python3.7/site-packages/sqlalchemy/util/compat.py", line 182, in raise_
    raise exception
AttributeError: Neither 'BindParameter' object nor 'Comparator' object has an attribute 'text'
```

Is filtering supported? 

Thanks for your time.

**Comment by aadel** [comment]
Filtering is supported but you hit an issue that occurred when Superset sends parametrized queries. Commit https://github.com/aadel/sqlalchemy-solr/commit/e95b756b3bf9161548c1e82199cc6b0541f9718d resolves this issue.

**Comment by anjackson** [comment]
Sorry for not replying before -- thank you for this!

**Comment by aadel** [comment]
Hello @anjackson! By now, another issue is fixed with filtration (d7a7b0bf2f6ece47953ab5c5190dc7222afd399b) and part of the related tests have been added (524fc40dbae3b85ada148de8cf54c0cf61440003).
